### ISSUE TITLE
[Documentation] 1.4 release is not backwards compatible

### ISSUE THREAD

**Issue by silviuburceadev** [tomitribe/http-signatures-java#37]
I was using 1.3 version and this constructor in my codebase
```java
public Signature(String keyId, String algorithm, String signature, List<String> headers) {
	this(keyId, getAlgorithm(algorithm), signature, headers);
```
When I upgraded to 1.4, the constructor is gone, from [this commit](https://github.com/tomitribe/http-signatures-java/commit/3eaa2c9c62d7b10c43c53b96a84ce69507da5d88#diff-39a89131a4a068beddcdeba8e4ab04bdL73)

I can see that `Signature` has a `parameterSpec` field now, for which you're passing a (extra) null in your unit tests. I don't mind adding an extra null in my codebase as well, but I'd have expected some release notes or bump the version to 2.0 to indicate a breaking change or keeping the constructor and deprecating it.

Do you have any documentation(apart from README) or a changelog that documents this which I've missed? If you don't, maybe we can start one.

**Comment by dnsmkl** [comment]
+1 vote on the issue

**Comment by jeanouii** [comment]
Hi,

Sorry for the delay responding this ticket.
I agree the backward compatibility issue could have been avoided.
Would you be interested in creating a small PR to put the constructor back and call the one with the parameter spec being null?

I would merge it straight.

Jlouis

**Comment by silviuburceadev** [comment]
Hi @jeanouii,

I can create a PR to bring back the constructor, but my time is very limited right now. I guess I should open a separate PR for the CHANGELOG.md file, right?

Best regards,
Silviu

**Comment by jeanouii** [comment]
Ideally yes. If it's easier for you do do a single PR, can you at least make sure the 2 tickets are mentioned in the PR?
I'd merge it anyways if you are wiling to contribute it.

**Comment by dblevins** [comment]
This should now be fixed and available in release 1.7.  If there is still an issue, please reopen.
